internal/providercache/package_install.go

```diff
@@ -26,6 +28,66 @@ import (
 // specific protocol and set of expectations.)
 var unzip = getter.ZipDecompressor{}
 
+const (
+	// httpClientRetryCountEnvName is the environment variable name used to customize
+	// the HTTP retry count for module downloads.
+	httpClientRetryCountEnvName = "TF_PROVIDER_DOWNLOAD_RETRY"
```

Comment:
I personally have an issue with how we are proliferating the amount of environment variables that are littered across the code. Environment variables are globally scoped and hurt testability. I would prefer not adding a new env variable.

@opentofu/core-engineers can you please take a look and help me out with an opinion here?

Reply:
Personally, I'm ok with this as it's a common pattern through the codebase.  As long as it is documented, I don't see an issue.

Reply:
@cam72cam It would be helpful if you could provide guidance on where to incorporate documentation for this.

Reply:
I would recommend adding documentation here: https://github.com/opentofu/opentofu/blob/main/website/docs/cli/config/environment-variables.mdx

Reply:
done